src/EntryPoints/Specials/SpecialManageApprovers.php

```diff
@@ -69,15 +67,13 @@ private function processCategoryAction( string $action, string $category, int $u
 		$currentCategories = $this->approverRepository->getApproverCategories( $userId );
 
 		switch ( $action ) {
+			case 'add-approver':
```

Comment:
1- We can remove the `add-approver` case. We can add the approver if it doesn't exist using `add` case, 

2-and also, the reason for moving it here now is that if the user accidentally adds the user that's already in the DB with many categories, it deletes all of the categories.

The reason for keeping `add-approver'` on the FE is to show the intent of the action/request. I can remove it though.